datatypes.go

```diff
@@ -115,6 +128,40 @@ type TriggerData struct {
 	Tags       []string `json:"__notifier_trigger_tags"`
 }
 
+func (trigger *TriggerData) PopulateDescription(events NotificationEvents) error {
```

Comment:
Probably better to not to edit current trigger with populated description and create a new trigger instead of it?